src/mlpack/methods/ann/layer/repeat_impl.hpp

```diff
@@ -0,0 +1,205 @@
+RepeatType<MatType>& RepeatType<MatType>::operator=(const RepeatType& other)
+{
+  if (this != &other)
+  {
+    Layer<MatType>::operator=(other);
+    multiples = other.multiples;
+    interleave = other.interleave;
+    outIdxs = other.outIdxs;
+    coefs = other.coefs;
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+RepeatType<MatType>& RepeatType<MatType>::operator=(RepeatType&& other) noexcept
+{
+  if (this != &other)
+  {
+    Layer<MatType>::operator=(std::move(other));
+    multiples = std::move(other.multiples);
+    interleave = std::move(other.interleave);
+    outIdxs = std::move(other.outIdxs);
+    coefs = std::move(other.coefs);
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+void RepeatType<MatType>::ComputeOutputDimensions()
+{
+  if (multiples.size() > this->inputDimensions.size())
+  {
+    std::ostringstream oss;
+    oss << "Repeat::ComputeOutputDimensions(): multiples vector must "
+        << "have the same or fewer dimensions than InputDimensions";
+    throw std::invalid_argument(oss.str());
+  }
+
+  size_t inputSize = this->inputDimensions[0];
+  for (size_t i = 1; i < this->inputDimensions.size(); i++)
+  {
+    inputSize *= this->inputDimensions[i];
+  }
+  arma::umat idxs = arma::regspace<arma::uvec>(0, inputSize-1);
+
+  // Here, we are going to pre-compute the source index for each output
+  // for a single tensor.  Since the tensors are flattened into 1-d
+  // vectors, we can fill the output row-wise based on these
+  // indices.
+  this->outputDimensions = this->inputDimensions;
+  size_t sizeMult = 1;
+  size_t outSize = 1;
+
+  if (interleave)
+  {
+    // iteratively reshape the index matrix such that the dimension
+    // to be repeated (and all prior) are flattened to a column, and
+    // then repelem rowwise.
+    for (size_t i = 0; i < multiples.size(); i++)
+    {
+      if (multiples[i] != 1)
+      {
+// For the first dimension, we need to do the repelem columnwise.
+        if (i == 0)
+        {
+          idxs.reshape(outSize * this->inputDimensions[i],
+                       idxs.n_elem / (outSize * this->inputDimensions[i]));
+          idxs = arma::repelem(idxs, multiples[i], 1);
+        } else
+        {
+          idxs.reshape(outSize,
+                       idxs.n_elem / outSize);
+          idxs = arma::repelem(idxs, 1, multiples[i]);
+        }
+        this->outputDimensions[i] *= multiples[i];
+        sizeMult *= multiples[i];
+      }
+      outSize *= this->outputDimensions[i];
+    }
+  }
+  else {
+    // iteratively reshape the index matrix such that the dimension
+    // to be replicated (and all prior) are flattened to a column, and
+    // then repmat columnwise.
+    for (size_t i = 0; i < multiples.size(); i++)
+    {
+      if (multiples[i] != 1)
+      {
+        idxs.reshape(outSize * this->inputDimensions[i],
+                     idxs.n_elem / (outSize * this->inputDimensions[i]));
+        idxs = arma::repmat(idxs, multiples[i], 1);
```

Comment:
It's up to you, but you could push down the `if (interleave)` all the way to here, since the only difference between the two branches is `arma::repmat` vs. `arma::repelem`.